.github/workflows/nix.yml

```
@@ -0,0 +1,29 @@
+name: nix
+on:
+  workflow_dispatch:
+  merge_group:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  build-nix-package:
+    strategy:
+      matrix:
+        os: [ubuntu-24.04, ubuntu-24.04-arm, macos-14]
```

```suggestion
        os: [ubuntu-24.04, ubuntu-22.04-arm, macos-14]
```

https://github.com/ChainSafe/forest/pull/5261

---

If we can't afford the minutes then let's not build the nix flake at all.

---

We can afford them, but I'm not convinced the usage of MacOS here is justified. Do you expect this build to pass on ARM and not on MacOS, especially given its deterministic setup? In the same way, we don't build and run RPC calibnet tests or unit tests on all worker types.

---

Generating cache artifacts on MacOS is quite useful for me since I do most of my development on MacOS. Perhaps a compromise would be to cache the compiled dependencies and releases of Forest.

---

Given your use case, would running the MacOS build only on `main` make sense, or would that cache be insufficient?

---

If we only run on 'main' then we risk red crosses.

---

So you expect the MacOS build to potentially fail even if the Linux one is green? I'm asking because my knowledge of `nix` is non-existent and I wouldn't expect such behaviour with, e.g., `docker build`, hence my original concern.

---

There are no guarantees that a build will succeed on MacOS, even if it was successful on Linux. I vote for building dependencies in PRs. This guarantees green checkmarks for `main` (since `main` will always see a populated cache).

---

Ok.